app/logic/util/util.ts

```diff
@@ -76,6 +76,16 @@ export function fileExtensionForMIMEType(mimetype: string) {
   return ".ext";
 }
 
+export async function throttleConnectionsPerSecond(nextConnectionTime: number[]) {
+  let milliseconds = Date.now();
+  nextConnectionTime.push(milliseconds + 1000);
+  milliseconds = nextConnectionTime.shift() - milliseconds;
+  if (milliseconds > 0) {
+    console.log(`Throttling for ${milliseconds}ms because there were ${nextConnectionTime.length} connections in the last second`);
+    await sleep(milliseconds / 1000);
+  }
+}
+
```

Comment:
util.js is only for very generic utility functions. Basically functions that should really be part of the web platform, but are not yet.